tests/unit/src/test/scala/tests/codeactions/StringActionsLspSuite.scala

```diff
@@ -0,0 +1,77 @@
+package tests.codeactions
+
+import scala.meta.internal.metals.codeactions.StringActions
+
+class MultlineStringLspSuite extends BaseCodeActionLspSuite("multilineString") {
+
+  check(
+    "empty-string",
+    """|package a
+       |
+       |object A {
+       |  val str = "<<>>"
+       |}
+       |""".stripMargin,
+    s"${StringActions.title}",
+    """|package a
+       |
+       |object A {
+       |  val str = '''|'''.stripMargin
+       |}
+       |""".stripMargin.replace("'", """"""")
+  )
+
+  check(
+    "string-selection",
+    """|package a
+       |
+       |object A {
+       |  val str = <<"this is a string">>
+       |}
+       |""".stripMargin,
+    s"${StringActions.title}",
+    """|package a
+       |
+       |object A {
+       |  val str = '''|this is a string'''.stripMargin
+       |}
+       |""".stripMargin.replace("'", """"""")
+  )
+
+  check(
+    "narrow-selection",
+    """|package a
+       |
+       |object A {
+       |  val str = "this i<<s >>a string"
+       |}
+       |""".stripMargin,
+    s"${StringActions.title}",
+    """|package a
+       |
+       |object A {
+       |  val str = '''|this is a string'''.stripMargin
+       |}
+       |""".stripMargin.replace("'", """"""")
+  )
+
+  check(
+    "interpolation-string",
+    """|package a
+       |
+       |object A {
+       |  val text = "text"
+       |  val str = s"<<this>> is an interpolation ${text}"
+       |}
+       |""".stripMargin,
+    s"${StringActions.title}",
+    """|package a
+       |
+       |object A {
+       |  val text = "text"
+       |  val str = s'''|this is an interpolation ${text}'''.stripMargin
+       |}
+       |""".stripMargin.replace("'", """"""")
+  )
+
```

Comment:
Let's add a test case with an already existing multiline string to check if we don't propose it flakily.

And maybe a test case with multiple strings on the same line?

Reply:
Thanks a lot for all your comments!
I've refactored everything. it looks better from my point of view :)

I just have a problem with multilines when testing. 
when I test that
```scala
"""|package a
   |
   |object A {
   |  val str = s'''|Hello
   |                |there'''.stripMargin + '''the <<cursor>> is actually here'''
   |}
   |""".stripMargin.replace("'", "\"")
```
it actually gives me that 
```scala
"""|package a
   |
   |object A {
   |  val str = s'''|Hello
   |there'''.stripMargin + '''the <<cursor>> is actually here'''
   |}
   |""".stripMargin.replace("'", "\"")
```
I'm already testing all the triple quotes cases so I'm not sure this test is useful

Reply:
Hmm... does it work in the editor? Maybe it's a problem with range formatting?

Reply:
I think we can skip this test then.

Reply:
okay we will skip that one, but we need some others :)